src/contracts/pods/EigenPodManager.sol

```diff
@@ -215,6 +225,38 @@ contract EigenPodManager is
         ownerToPod[podOwner].withdrawRestakedBeaconChainETH(destination, shares);
     }
 
+    /// @notice Called by proving service to fulfill a partial withdrawal proof request
+    function proofServiceCallback(
+        bytes32 blockRoot,
+        uint64 oracleTimestamp,
+        WithdrawalCallbackInfo[] calldata callbackInfo
+    ) external onlyProofService proofServiceEnabled nonReentrant {
+        require(blockRoot == getBlockRootAtTimestamp(oracleTimestamp), "EigenPodManager.proofServiceCallback: block root does not match oracleRoot for that timestamp");
+        for(uint256 i = 0; i < callbackInfo.length; i++) {
+            // these checks are verified in the snark, we add them here again as a sanity check
+            require(oracleTimestamp >= callbackInfo[i].endTimestamp, "EigenPodManager.proofServiceCallback: oracle timestamp must be greater than or equal to callback timestamp");
+            require(callbackInfo[i].fee <= callbackInfo[i].maxFee, "EigenPod.fulfillPartialWithdrawalProofRequest: fee must be less than or equal to maxFee");
+            IEigenPod pod = ownerToPod[callbackInfo[i].podOwner];
+            require(address(pod) != address(0), "EigenPodManager.proofServiceCallback: pod does not exist");
+            pod.fulfillPartialWithdrawalProofRequest(callbackInfo[i], proofService.feeRecipient);
+        }
+    }
+
+    /// @notice enables partial withdrawal proving via offchain proofs
+    function enableProofService() external onlyOwner {
+        require(!partialWithdrawalProofSwitch);
```

Comment:
Do we want an error message here? I'm fine without it but we do typically have one.

Reply:
added